src/common/post.ts

```diff
@@ -252,3 +257,80 @@ export const validatePost = (
 export const submitArticleThreshold = parseInt(
   process.env.SUBMIT_ARTICLE_THRESHOLD,
 );
+
+export const insertCodeSnippets = async ({
+  entityManager,
+  post,
+  codeSnippetsJson,
+}: {
+  entityManager: EntityManager;
+  post: Pick<Post, 'id'>;
+  codeSnippetsJson: PostCodeSnippetJsonFile;
+}) => {
+  const codeSnippets = codeSnippetsJson.snippets.reduce(
+    (acc, codeSnippetContent, index) => {
+      const checksum = createHash('sha1');
+      checksum.update(codeSnippetContent);
+      const contentHash = checksum.digest('hex');
+
+      if (!acc.has(contentHash)) {
+        acc.set(
+          contentHash,
+          entityManager.getRepository(PostCodeSnippet).create({
+            postId: post.id,
+            contentHash,
+            order: index,
+            content: codeSnippetContent,
+          }),
+        );
+      }
+
+      return acc;
+    },
+    new Map<string, PostCodeSnippet>(),
+  );
+
+  const uniqueCodeSnippets = Array.from(codeSnippets.values());
```

Comment:
We also have uniqueifyArray not sure it works here :)

Reply:
it works over set with primitive values and I am doing objects here, I can extend that function to support objects as well?

Reply:
I see :).
Hmm yeah up to you, but `uniqueifyObjectArray` would also be nice to have. think we do it more often.

Reply:
fc2466fb445097bb5b5cec13fdf1ab4696cf1712